src/ng/filter/orderBy.js

```diff
@@ -176,88 +176,110 @@
 orderByFilter.$inject = ['$parse'];
 function orderByFilter($parse) {
   return function(array, sortPredicate, reverseOrder) {
+
     if (!(isArrayLike(array))) return array;
-    sortPredicate = isArray(sortPredicate) ? sortPredicate : [sortPredicate];
+
+    if (!isArray(sortPredicate)) { sortPredicate = [sortPredicate]; }
     if (sortPredicate.length === 0) { sortPredicate = ['+']; }
-    sortPredicate = sortPredicate.map(function(predicate) {
-      var descending = false, get = predicate || identity;
-      if (isString(predicate)) {
+
+    var predicates = processPredicates(sortPredicate);
+    var compareValues = Array.prototype.map.call(array, getComparisonObject);
+
+    compareValues.sort(doComparison);
+    array = compareValues.map(function(item) { return item.value; });
+    if (reverseOrder) { array.reverse(); }
+    return array;
+
+    function getComparisonObject(value, index) {
+      return {
+        value: value,
+        predicateValues: predicates.map(function(predicate) {
+          return getPredicateValue(predicate.get(value), index);
+        })
+      };
+    }
+
+    function doComparison(v1, v2) {
+      return predicates.reduce(function(result, predicate, index) {
```

Comment:
yes --- the spec mandates `HasProperty` checks for each item which are slow almost everywhere, and can't be optimized away because the callback can manipulate the array